src/qat/purr/compiler/runtime.py

```diff
@@ -252,6 +259,56 @@ def fexecute(instrs):
         )
 
 
+class NewQuantumRuntime(QuantumRuntime, InvokerMixin):
+    """
+    Uses the new pass infrastructure.
+
+    Notice how polymorphic calls to XEngine.optimize() and XEngine.validate() are avoided. Instead, we have
+    a flat structure of passes. This allows developers to focus on efficiently implementing a pass and easily test,
+    demonstrate, and register passes without worrying too much about where it fits into the global compilation
+    workflow.
```

Comment:
👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻👍🏻

Reply:
This probably answers my questions above, correct?

Reply:
Yep